### PR TITLE
Account for Logstash memory in resource aggregator

### PR DESCRIPTION
**Pull request by pebrc**
Logstash resources should be counted towards the ECK managed memory and ERU usage.

### CODE REVIEW

File: pkg/license/testdata/stack.yaml [hunk 1]
```diff
@@ -143,4 +143,24 @@ spec:
...
+apiVersion: logstash.k8s.elastic.co/v1alpha1
+kind: Logstash
+metadata:
+  name: logstash-sample
+spec:
+  count: 1
+  podTemplate:
+    spec:
+      containers:
+        - name: enterprise-search
```
Comment (barkbay) [inline pkg/license/testdata/stack.yaml:157]:
```suggestion
        - name: logstash
```

File: pkg/license/testdata/stack.yaml [hunk 2]
```diff
@@ -143,4 +143,24 @@ spec:
...
+          resources:
+            requests:
+              cpu: 3
+              memory: 1Gi
+            limits:
+              memory: 4Gi
+          env:
+            - name: LS_JAVA_OPTS
+              value: -Xms2g -Xmx2g
```
Comment (barkbay) [inline pkg/license/testdata/stack.yaml:166]: Let's maybe use a different value than the default one?

**Review by barkbay — APPROVED** [review]
I think the test should be fixed, LGTM otherwise.

**Review by robbavey — APPROVED** [review]
LGTM

**Comment by pebrc** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|2.13|[](https://github.com/elastic/cloud-on-k8s/pull/7854)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Pull request merged**